### PR TITLE
DAOS-2205 drpc: Provide more useful failure codes

### PR DESCRIPTION
**Pull request by kjacque**
This patch adds error codes for some additional common dRPC
failure cases that were previously just reported as "Failure."
It also standardizes usage between the C and Go dRPC server
implementations.

- Regenerate dRPC Protobuf files.
- Add generated files to checkpatch ignore list.
- Update mgmt svc dRPC handlers in Data Plane to use the new
  codes.
- Add drpc.Failure rich error type in Control Plane.
- Change dRPC Go server and modules to use new Failures to
  communicate common errors.
- Respond to bad call in C dRPC server instead of just hanging
  up.
- Update drpc_response_create() to accept NULL to respond to
  bad calls.

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

File: src/include/daos/drpc.pb-c.h [hunk 1]
```diff
@@ -60,16 +94,22 @@ struct  _Drpc__Call
...
  */
 struct  _Drpc__Response
 {
   ProtobufCMessage base;
+  /*
+   * Sequence number of the Call that triggered this response.
+   */
   int64_t sequence;
+  /*
+   * High-level status of the RPC. If SUCCESS, method-specific status may be included in the body.
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/include/daos/drpc.pb-c.h]: Checkpatch should be ignoring generated files. Will add this one to Jenkinsfile.

File: src/common/drpc.pb-c.c [hunk 2]
```diff
@@ -238,20 +238,26 @@ const ProtobufCMessageDescriptor drpc__response__descriptor =
...
-static const ProtobufCEnumValue drpc__status__enum_values_by_number[5] =
+static const ProtobufCEnumValue drpc__status__enum_values_by_number[8] =
 {
   { "SUCCESS", "DRPC__STATUS__SUCCESS", 0 },
   { "SUBMITTED", "DRPC__STATUS__SUBMITTED", 1 },
   { "FAILURE", "DRPC__STATUS__FAILURE", 2 },
   { "UNKNOWN_MODULE", "DRPC__STATUS__UNKNOWN_MODULE", 3 },
   { "UNKNOWN_METHOD", "DRPC__STATUS__UNKNOWN_METHOD", 4 },
+  { "INVALID_MESSAGE", "DRPC__STATUS__INVALID_MESSAGE", 5 },
+  { "INVALID_PAYLOAD", "DRPC__STATUS__INVALID_PAYLOAD", 6 },
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/common/drpc.pb-c.c]: This is a good point. I was originally thinking the "invalid" messages were more helpful. But as I think about it more, it occurs to me that, while an unmarshaling failure usually means that the data format was invalid, that isn't necessarily the case (could be a memory allocation failure under the covers, for example). I will change these. Thanks.

File: src/control/drpc/failure.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+// UnmarshalingPayloadFailure creates a Failure for a failed attempt to unmarshal a call payload.
+func UnmarshalingPayloadFailure() Failure {
+	return NewFailure(Status_FAILED_UNMARSHAL_PAYLOAD)
+}
+
+// MarshalingFailure creates a Failure for a failed attempt at marshaling a response.
+func MarshalingFailure() Failure {
+	return NewFailure(Status_FAILED_MARSHAL)
+}
```
Comment (mjmac) [inline src/control/drpc/failure.go]: My initial reaction to these was to wonder why you didn't just declare them as variables... It'd be better to make them consts, of course, but go doesn't support non-numeric consts. I guess doing it this way ensures that there are no shenanigans as a result of messing with the global variables?

Comment (kjacque) [reply src/control/drpc/failure.go]: Yeah. Without the ability to make them consts, it seemed like a bad idea to make them global variables. Could go either way, I guess, but I had the NewFailure factory method already anyway...

File: src/control/drpc/failure.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	case Status_INVALID_MESSAGE:
+		return "invalid incoming message"
+	case Status_INVALID_PAYLOAD:
+		return "invalid method-specific payload"
+	case Status_FAILED_MARSHAL:
+		return "failed to marshal response payload"
+	case Status_SUCCESS:
+		fallthrough
+	case Status_SUBMITTED:
+		return ""
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/control/drpc/failure.go]: Good catch, I'll simplify it if something else comes up that needs to change.

**Pull request merged**